### PR TITLE
feat(pm_auth): Added pm_auth_config to merchant_connector_account

### PR DESCRIPTION
**Pull request by Sarthak1799**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

We will treat pm_auth services as connectors hence requiring a change in the MCA table.
The pm_auth will be of connector type = "payment_method_auth".
Additionally, a new column called pm_auth_config will relate MCA records to their respective chosen auth services, based on payment_method and pmt. Should be configured in dashboard.

VAS conjugate PR - https://github.com/juspay/hyperswitch-cloud/pull/2191
Updated VAS conjugate PR - https://github.com/juspay/hyperswitch-cloud/pull/2172

### Additional Changes

- [ ] This PR modifies the API contract
- [x] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Configure a pm auth connector - 
`curl --location --request POST 'http://localhost:8080/account/sarthak1/connectors' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: test_admin' \
--data-raw '{
    "connector_type": "payment_method_auth",
    "connector_name": "plaid",
     "business_country": "US",
    "business_label": "default",
    "connector_account_details": {
        "auth_type": "BodyKey",
        "api_key": "some_key",
        "key1": "secret"
    },
    "test_mode": false,
    "disabled": false
}'`

configure a PG connector with pm_auth_config provided - 
`curl --location --request POST 'http://localhost:8080/account/sarthak1/connectors' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: test_admin' \
--data-raw '{
    "connector_type": "fiz_operations",
    "connector_name": "stripe",
    "business_country": "US",
    "business_label": "default",
    "connector_account_details": {
        "auth_type": "HeaderKey",
        "api_key": "some_key"
    },
    "test_mode": false,
    "disabled": false,
    "payment_methods_enabled": [
        {
            "payment_method": "card",
            "payment_method_types": [
                {
                    "payment_method_type": "credit",
                    "card_networks": [
                        "AmericanExpress",
                        "DinersClub",
                        "Discover",
                        "JCB",
                        "Mastercard",
                        "Visa"
                    ],
                    "minimum_amount": 1,
                    "maximum_amount": 68607706,
                    "recurring_enabled": true,
                    "installment_payment_enabled": true,
                    "accepted_currencies": {
                        "type": "enable_only",
                        "list": [
                            "CZK",
                            "RON",
                            "TRY",
                            "USD",
                            "GBP",
                            "CAD",
                            "AUD",
                            "ARS",
                            "CHF",
                            "CNY",
                            "DKK",
                            "GHS",
                            "JPY",
                            "PLN",
                            "NGN",
                            "NOK",
                            "KES",
                            "SEK",
                            "NZD",
                            "ZAR",
                            "SGD",
                            "INR"
                        ]
                    }
                },
                {
                    "payment_method_type": "debit",
                    "card_networks": [
                        "AmericanExpress",
                        "DinersClub",
                        "Discover",
                        "JCB",
                        "Mastercard",
                        "Visa"
                    ],
                    "minimum_amount": 1,
                    "maximum_amount": 68607706,
                    "recurring_enabled": true,
                    "installment_payment_enabled": true,
                    "accepted_currencies": {
                        "type": "enable_only",
                        "list": [
                            "CZK",
                            "RON",
                            "TRY",
                            "USD",
                            "GBP",
                            "CAD",
                            "AUD",
                            "ARS",
                            "CHF",
                            "CNY",
                            "DKK",
                            "GHS",
                            "JPY",
                            "PLN",
                            "NGN",
                            "NOK",
                            "KES",
                            "SEK",
                            "NZD",
                            "ZAR",
                            "SGD",
                            "INR"
                        ]
                    }
                },
                {
                    "payment_method_type": "credit",
                    "card_networks": [
                        "Mastercard",
                        "Visa"
                    ],
                    "minimum_amount": 1,
                    "maximum_amount": 68607706,
                    "recurring_enabled": true,
                    "installment_payment_enabled": true,
                    "accepted_currencies": {
                        "type": "enable_only",
                        "list": [
                            "EUR"
                        ]
                    }
                },
                {
                    "payment_method_type": "debit",
                    "card_networks": [
                        "Mastercard",
                        "Visa"
                    ],
                    "minimum_amount": 1,
                    "maximum_amount": 68607706,
                    "recurring_enabled": true,
                    "installment_payment_enabled": true,
                    "accepted_currencies": {
                        "type": "enable_only",
                        "list": [
                            "EUR"
                        ]
                    }
                }
            ]
        },
        {
            "payment_method": "bank_redirect",
            "payment_method_types": [
                {
                    "payment_method_type": "eps",
                    "recurring_enabled": true,
                    "installment_payment_enabled": true,
                    "accepted_currencies": {
                        "type": "enable_only",
                        "list": [
                            "EUR"
                        ]
                    }
                },
                {
                    "payment_method_type": "giropay",
                    "recurring_enabled": true,
                    "installment_payment_enabled": true,
                    "accepted_currencies": {
                        "type": "enable_only",
                        "list": [
                            "EUR"
                        ]
                    }
                },
                {
                    "payment_method_type": "ideal",
                    "recurring_enabled": true,
                    "installment_payment_enabled": true,
                    "accepted_currencies": {
                        "type": "enable_only",
                        "list": [
                            "EUR"
                        ]
                    }
                }
            ]
        },
        {
            "payment_method": "wallet",
            "payment_method_types": [
                {
                    "payment_method_type": "apple_pay",
                    "payment_experience": "invoke_sdk_client",
                    "card_networks": null,
                    "accepted_countries": null,
                    "minimum_amount": 1,
                    "maximum_amount": 68607706,
                    "recurring_enabled": true,
                    "installment_payment_enabled": true,
                    "accepted_currencies": {
                        "type": "enable_only",
                        "list": [
                            "USD",
                            "GBP",
                            "CAD",
                            "AUD",
                            "EUR"
                        ]
                    }
                }
            ]
        }
    ],
    "metadata": {
        "city": "NY",
        "unit": "245"
    },
    "pm_auth_config": {
        "pm_auth_configs": [
            {
                "payment_method": "card",
                "payment_method_type": "credit",
                "auth_service": "plaid_US_default"
            }
        ]
    }
}'`

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by vspecky — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

File: crates/api_models/src/admin.rs [hunk 1]
```diff
@@ -633,6 +633,8 @@ pub struct MerchantConnectorCreate {
     pub connector_webhook_details: Option<MerchantConnectorWebhookDetails>,
     /// Identifier for the business profile, if not provided default will be chosen from merchant account
     pub profile_id: Option<String>,
+
+    pub pm_auth_config: Option<serde_json::Value>,
```
Comment (Chethan-rao) [inline crates/api_models/src/admin.rs:637]: Could u plz add a doc comment. can be taken up in another PR

File: crates/api_models/src/admin.rs [hunk 2]
```diff
@@ -736,6 +738,8 @@ pub struct MerchantConnectorResponse {
     pub profile_id: Option<String>,
     /// identifier for the verified domains of a particular connector account
     pub applepay_verified_domains: Option<Vec<String>>,
+
+    pub pm_auth_config: Option<serde_json::Value>,
```
Comment (Chethan-rao) [inline crates/api_models/src/admin.rs:742]: Same here, doc comment

File: crates/api_models/src/admin.rs [hunk 3]
```diff
@@ -805,6 +809,8 @@ pub struct MerchantConnectorUpdate {
         }
     }))]
     pub connector_webhook_details: Option<MerchantConnectorWebhookDetails>,
+
+    pub pm_auth_config: Option<serde_json::Value>,
```
Comment (Chethan-rao) [inline crates/api_models/src/admin.rs:813]: Doc comment

File: crates/router/src/core/admin.rs [hunk 4]
```diff
@@ -682,6 +682,7 @@ pub async fn create_payment_connector(
         },
         profile_id: Some(profile_id.clone()),
         applepay_verified_domains: None,
+        pm_auth_config: req.pm_auth_config.clone(),
```
Comment (Chethan-rao) [inline crates/router/src/core/admin.rs:685]: is this clone required?
